lib/fsi.go

```diff
@@ -30,16 +30,18 @@ func (m FSIMethods) Name() string {
 // Attributes defines attributes for each method
 func (m FSIMethods) Attributes() map[string]AttributeSet {
 	return map[string]AttributeSet{
-		"createlink":            {AEFSICreateLink, "POST"},
-		"unlink":                {AEFSIUnlink, "POST"},
-		"status":                {AEStatus, "GET"},
-		"whatchanged":           {AEWhatChanged, "GET"},
+		"status":                {AEStatus, "POST"},
+		"caninitdatasetworkdir": {AECanInitDatasetWorkDir, "POST"},
+		"init":                  {AEInit, "POST"},
 		"checkout":              {AECheckout, "POST"},
-		"write":                 {AEFSIWrite, "POST"},
 		"restore":               {AERestore, "POST"},
-		"init":                  {AEInit, "POST"},
-		"caninitdatasetworkdir": {AECanInitDatasetWorkDir, "GET"},
-		"ensureref":             {AEEnsureRef, "POST"},
+		"write":                 {AEFSIWrite, "POST"},
+		"createlink":            {AEFSICreateLink, "POST"},
+		"unlink":                {AEFSIUnlink, "POST"},
+
+		// TODO(arqu): reorganize these handlers into appropriate packages
+		"whatchanged": {AEWhatChanged, "POST"}, // dataset.componentstatus
+		"ensureref":   {AEEnsureRef, "POST"},   // ??
```

Comment:
Unsure what the purpose of `ensureref` is.

Reply:
EnsureRef is used by FSI to fixup linked datasets that may have been modified by outside programs (or manual user intervention) between runs of qri. For example, if someone checks out a dataset to a working directory, and then deletes the `.qr-ref` file from the folder it was checked out to, when another command is run, EnsureRef will notice this and unlink the dataset. It needs to be available on RPC because most commands will call it via `cmd/ref_select.go`. It runs the method `fsi.EnsureRef`.

Ah, and I just noticed now it's not in the spec file. We should add it there.

Reply:
added!